cf_xarray/geometry.py

```diff
@@ -30,6 +33,122 @@
 # 1. node coordinates are exact; the 'normal' coordinates are a reasonable value to use, if you do not know how to interpret the nodes.
 
 
+@dataclass
+class GeometryNames:
+    """Helper class to ease handling of all the variable names needed for CF geometries."""
+
+    def __init__(
+        self,
+        suffix: str = "",
+        grid_mapping_name: str | None = None,
+        grid_mapping: str | None = None,
+    ):
+        self.container_name: str = GEOMETRY_CONTAINER_NAME + suffix
+        self.node_dim: str = "node" + suffix
+        self.node_count: str = "node_count" + suffix
+        self.node_coordinates_x: str = "x" + suffix
+        self.node_coordinates_y: str = "y" + suffix
+        self.coordinates_x: str = "crd_x" + suffix
+        self.coordinates_y: str = "crd_y" + suffix
+        self.part_node_count: str = "part_node_count" + suffix
+        self.part_dim: str = "part" + suffix
+        self.interior_ring: str = "interior_ring" + suffix
+        self.attrs_x: dict[str, str] = {}
+        self.attrs_y: dict[str, str] = {}
+
+        gmattr = {"grid_mapping": grid_mapping} if grid_mapping else {}
+        # Special treatment of selected grid mappings
+        if grid_mapping_name in ["latitude_longitude", "rotated_latitude_longitude"]:
+            # Special case for longitude_latitude type grid mappings
+            self.coordinates_x = "lon"
+            self.coordinates_y = "lat"
+            if grid_mapping_name == "latitude_longitude":
+                self.attrs_x = dict(
+                    units="degrees_east", standard_name="longitude", **gmattr
+                )
+                self.attrs_y = dict(
+                    units="degrees_north", standard_name="latitude", **gmattr
+                )
+            elif grid_mapping_name == "rotated_latitude_longitude":
+                self.attrs_x = dict(
+                    units="degrees_east", standard_name="grid_longitude", **gmattr
+                )
+                self.attrs_y = dict(
+                    units="degrees_north", standard_name="grid_latitude", **gmattr
+                )
+        elif grid_mapping_name is not None:
+            self.attrs_x = dict(standard_name="projection_x_coordinate", **gmattr)
+            self.attrs_y = dict(standard_name="projection_y_coordinate", **gmattr)
+
+    @property
+    def geometry_container_attrs(self) -> dict[str, str]:
+        return {
+            "node_count": self.node_count,
+            "node_coordinates": f"{self.node_coordinates_x} {self.node_coordinates_y}",
+            "coordinates": f"{self.coordinates_x} {self.coordinates_y}",
+        }
+
+    def coords(self, *, dim: str, x, y, crdX, crdY) -> dict[str, xr.DataArray]:
```

Comment:
TODO: save `dim` as `self.core_dim`? This is the core dimension for the decoded shapely geometries.